docs/source/cpp/gdb.rst

```diff
@@ -0,0 +1,167 @@
+
+You will have to ``source`` it on each new GDB session.  You might want to
+make this implicit by adding the ``source`` invocation in a
+`gdbinit <https://sourceware.org/gdb/onlinedocs/gdb/gdbinit-man.html>`_ file.
+
+
+Automatic loading
+-----------------
+
+GDB provides a facility to automatically load scripts or extensions for each
+object file or library that is involved in a debugging session.  You will need
+to:
+
+1. Find out what the *auto-load* location(s) is/are for your GDB install.
+   This can be determined using ``show`` subcommands on the GDB prompt;
+   the answer will depend on the operating system.
+
+   Here is an example on Ubuntu::
+
+      (gdb) show auto-load scripts-directory
+      List of directories from which to load auto-loaded scripts is $debugdir:$datadir/auto-load.
+      (gdb) show data-directory
+      GDB's data directory is "/usr/share/gdb".
+      (gdb) show debug-file-directory
+      The directory where separate debug symbols are searched for is "/usr/lib/debug".
+
+   This tells you that the directories used for auto-loading are
+   ``$debugdir`` and ``$datadir/auto-load``, which expand to
+   ``/usr/lib/debug/`` and ``/usr/share/gdb/auto-load`` respectively.
+
+2. Find out the full path to the Arrow C++ DLL, *with all symlinks resolved*.
+   For example, you might have installed Arrow 7.0 in ``/usr/local`` and the
+   path to the Arrow C++ DLL could then be ``/usr/local/lib/libarrow.so.700.0.0``.
+
+3. Determine the actual auto-load script path.  It is computed by *a)* taking
+   the path of the auto-load directory of your choice, *b)* appending the full
+   path to the Arrow C++ DLL, *c)* appending ``-gdb.py`` at the tail.
+
+   In the example above, if we choose ``/usr/share/gdb/auto-load`` as auto-load
+   directory, the full path to the auto-load script will have to be
+   ``/usr/share/gdb/auto-load/usr/local/lib/libarrow.so.700.0.0-gdb.py``.
+
+4. Either copy or symlink the `GDB extension`_ to the file path determined
+   in step 3 above.
+
+If everything went well, then as soon as GDB encounters the Arrow C++ DLL,
+it will automatically load the Arrow GDB extension so as to pretty-print
+Arrow C++ classes on the display prompt.
+
+
+Supported classes
+-----------------
+
+The Arrow GDB extension provides pretty-printing for the core Arrow C++ classes:
+
+* :class:`arrow::DataType` and subclasses
+* :class:`arrow::Field`, :class:`arrow::Schema` and :class:`arrow::KeyValueMetadata`
+* :class:`arrow::ArrayData`, :class:`arrow::Array` and subclasses
+* :class:`arrow::Scalar` and subclasses
+* :class:`arrow::ChunkedArray`, :class:`arrow::RecordBatch` and :class:`arrow::Table`
+* :class:`arrow::Datum`
+
+Important utility classes are also covered:
+
+* :class:`arrow::Status` and :class:`arrow::Result`
+* :class:`arrow::Buffer` and subclasses
+* :class:`arrow::Decimal128`, :class:`arrow::Decimal256`
+* :class:`arrow::util::string_view`, :class:`arrow::util::optional`,
+  :class:`arrow::util::Variant`
```

Comment:
We don't have any links for `arrow::Result`, `arrow::util::string_view`, `arrow::util::optional`, or `arrow::util::Variant` (presumably because they aren't contained in some .rst file somewhere)

Reply:
That seems fine.